railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -88,26 +91,25 @@ def reference?(type)
         private
           # parse possible attribute options like :limit for string/text/binary/integer, :precision/:scale for decimals or :polymorphic for references/belongs_to
           # when declaring options curly brackets should be used
-          def parse_type_and_options(type_with_possible_options)
-            return type_with_possible_options, {} unless type_with_possible_options&.match?(TYPE_WITH_OPTIONS_RE)
-
-            case type_with_possible_options
-            when /(string|text|binary|integer)\{(\d+)\}/
-              return $1, limit: $2.to_i
-            when /decimal\{(\d+)[,.-](\d+)\}/
-              return :decimal, precision: $1.to_i, scale: $2.to_i
+          def parse_type_and_options(type, options)
+            return type, {} if options.blank?
+
+            if type.presence_in(%w(string text binary integer)) && options.match?(/^\d+$/)
+              return type.to_sym, limit: options.to_i
```

Comment:
Why do we need to call `to_sym` here, seems like the other branches don't?

Reply:
True. 21a0a75a8c